File: salmon_king/data/events.py
"""Event table. Mammal behavior matches the fishery, not a generic tycoon."""

from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class EventDef:
    id: str
    weight: float
    title: str
    kind: str  # weather, mammal, mechanical, market, adfg, camp, crew


EVENT_TABLE: tuple[EventDef, ...] = (
    EventDef("williwaw", 1.1, "Williwaw", "weather"),
    EventDef("advection_fog", 1.0, "Advection fog", "weather"),
    EventDef("shelikof_seas", 0.9, "Shelikof seas", "weather"),
    EventDef("whale_pass_rip", 0.35, "Whale Passage rip", "weather"),
    EventDef("september_winter", 0.5, "September weather", "weather"),
    EventDef("sea_lion_raid", 1.2, "Steller sea lions", "mammal"),
    EventDef("harbor_seal_raid", 0.8, "Harbor seals", "mammal"),
    EventDef("otter_foul", 0.45, "Sea otter / gear foul", "mammal"),
    EventDef("orca_residents", 0.35, "Resident killer whales", "mammal"),
    EventDef("orca_transients", 0.30, "Transient killer whales", "mammal"),
    EventDef("porpoise_bycatch", 0.08, "Harbor porpoise", "mammal"),
    EventDef("humpback_wrap", 0.12, "Humpback", "mammal"),
    EventDef("outboard_down", 0.55, "Outboard down", "mechanical"),
    EventDef("net_tear", 0.70, "Torn net", "mechanical"),
    EventDef("skiff_swamped", 0.25, "Skiff swamped", "mechanical"),
    EventDef("ice_failure", 0.40, "Ice / RSW failure", "mechanical"),
    EventDef("tender_late", 0.55, "Tender late", "market"),
    EventDef("price_cut", 0.40, "Price cut", "market"),
    EventDef("cannery_dark", 0.22, "Cannery dark", "market"),
    EventDef("adfg_extension", 0.35, "ADF&G extension", "adfg"),
    EventDef("adfg_pulse", 0.30, "ADF&G pulse", "adfg"),
    EventDef("food_spoil", 0.28, "Food spoilage", "camp"),
    EventDef("crew_fight", 0.30, "Crew fight", "crew"),
    EventDef("bonus_school", 0.45, "Fish showing", "adfg"),
)

File: src/data/events.rs
//! Event table. Mammal behavior matches the fishery, not a generic tycoon.

#[derive(Clone, Copy, Debug)]
pub struct EventDef {
    pub id: &'static str,
    pub weight: f64,
    pub title: &'static str,
    pub kind: &'static str,
}

pub const EVENT_TABLE: [EventDef; 24] = [
    EventDef { id: "williwaw", weight: 1.1, title: "Williwaw", kind: "weather" },
    EventDef { id: "advection_fog", weight: 1.0, title: "Advection fog", kind: "weather" },
    EventDef { id: "shelikof_seas", weight: 0.9, title: "Shelikof seas", kind: "weather" },
    EventDef { id: "whale_pass_rip", weight: 0.35, title: "Whale Passage rip", kind: "weather" },
    EventDef { id: "september_winter", weight: 0.5, title: "September weather", kind: "weather" },
    EventDef { id: "sea_lion_raid", weight: 1.2, title: "Steller sea lions", kind: "mammal" },
    EventDef { id: "harbor_seal_raid", weight: 0.8, title: "Harbor seals", kind: "mammal" },
    EventDef { id: "otter_foul", weight: 0.45, title: "Sea otter / gear foul", kind: "mammal" },
    EventDef { id: "orca_residents", weight: 0.35, title: "Resident killer whales", kind: "mammal" },
    EventDef { id: "orca_transients", weight: 0.30, title: "Transient killer whales", kind: "mammal" },
    EventDef { id: "porpoise_bycatch", weight: 0.08, title: "Harbor porpoise", kind: "mammal" },
    EventDef { id: "humpback_wrap", weight: 0.12, title: "Humpback", kind: "mammal" },
    EventDef { id: "outboard_down", weight: 0.55, title: "Outboard down", kind: "mechanical" },
    EventDef { id: "net_tear", weight: 0.70, title: "Torn net", kind: "mechanical" },
    EventDef { id: "skiff_swamped", weight: 0.25, title: "Skiff swamped", kind: "mechanical" },
    EventDef { id: "ice_failure", weight: 0.40, title: "Ice / RSW failure", kind: "mechanical" },
    EventDef { id: "tender_late", weight: 0.55, title: "Tender late", kind: "market" },
    EventDef { id: "price_cut", weight: 0.40, title: "Price cut", kind: "market" },
    EventDef { id: "cannery_dark", weight: 0.22, title: "Cannery dark", kind: "market" },
    EventDef { id: "adfg_extension", weight: 0.35, title: "ADF&G extension", kind: "adfg" },
    EventDef { id: "adfg_pulse", weight: 0.30, title: "ADF&G pulse", kind: "adfg" },
    EventDef { id: "food_spoil", weight: 0.28, title: "Food spoilage", kind: "camp" },
    EventDef { id: "crew_fight", weight: 0.30, title: "Crew fight", kind: "crew" },
    EventDef { id: "bonus_school", weight: 0.45, title: "Fish showing", kind: "adfg" },
];
